versioned/spi/src/main/java/org/projectnessie/versioned/NamedReferencesParams.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned;
+
+import javax.annotation.Nullable;
+import org.immutables.value.Value;
+
+/**
+ * Parameters that control the values that shall be returned via for each {@link NamedReference} via
+ * functionality to retrieve the named references.
+ */
+@Value.Immutable
+public interface NamedReferencesParams {
+
+  NamedReferencesParams DEFAULT = builder().build();
+
+  static ImmutableNamedReferencesParams.Builder builder() {
+    return ImmutableNamedReferencesParams.builder();
+  }
+
+  /**
+   * Name of the default branch to use as a reference for the computation of {@link
+   * NamedReference#getAheadBehind()}. If this parameter is not {@code null}, the branch must exist.
+   *
+   * @return name of the default branch. Can be {@code null}, if not needed.
+   */
+  @Nullable
+  String getDefaultBranchName();
+
+  /**
+   * Whether to compute {@link NamedReference#getAheadBehind()}, requires a non-{@code null} {@link
+   * #getDefaultBranchName()}, defaults to {@code false}.
+   */
+  @Value.Default
+  default boolean isComputeAheadBehind() {
+    return false;
+  }
+
+  /** Whether to identify the common ancestor on the default branch. */
+  @Value.Default
+  default boolean isComputeCommonAncestor() {
+    return false;
+  }
+
+  /**
+   * Whether to retrieve the commit-meta information of each reference's HEAD commit in {@link
+   * NamedReference#getHeadCommitMeta()}, defaults to {@code false}.
+   */
+  @Value.Default
+  default boolean isRetrieveHead() {
```

Comment:
maybe `isRetrieveCommitMetaForHead`? That's at least more understandable and wouldn't force people to use the javadoc to understand what this method is about ;)